code/mob/living/critter/flockcritter_parent.dm

```diff
@@ -114,6 +114,31 @@
 	if (istype(target, /obj/item) && target.loc == src) //no batong for radio birds
 		target.emp_act()
 
+/mob/living/critter/flock/weapon_attack(atom/target, obj/item/W, reach, params)
+	if (HAS_ATOM_PROPERTY(target, PROP_ATOM_FLOCK_THING) && target.density)
+		if (!reach && istype(W, /obj/item/gun))
+			boutput(src, "<span class='alert'>The grip tool refuses to harm this, jamming briefly.</span>")
+			return
+		return ..()
+	if (istype(W, /obj/item/gun))
+		if (src.a_intent == INTENT_GRAB && reach)
+			boutput(src, "<span class='alert'>You can't gun grab someone!</span>")
```

Comment:
Not a fan of meta-game messages, maybe just bonk the target with the gun instead?

Reply:
Now does a point blank shoot